firstvoices/backend/views/sites_views.py

```diff
@@ -96,3 +96,76 @@ def list(self, request, *args, **kwargs):
         ]
 
         return Response(data)
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="A list of language sites a given user is a member of, "
+        "grouped by language. If there are no accessible sites or "
+        "the user does not have membership to any site then the "
+        "list will be empty. Sites with no specified language "
+        "will be grouped under 'Other'.",
+        responses={
+            200: inline_serializer(
+                name="InlineLanguageSerializer",
+                fields={
+                    "language": serializers.CharField(),
+                    "sites": SiteSummarySerializer(many=True),
+                },
+            ),
+        },
+    ),
+)
+class MySitesViewSet(
+    AutoPermissionViewSetMixin, mixins.ListModelMixin, viewsets.GenericViewSet
+):
+    """
+    Summary information about language sites a given user has membership to.
+    """
+
+    http_method_names = ["get"]
+    pagination_class = None
+    serializer_class = SiteSummarySerializer
+
+    def get_queryset(self):
+        # get memberships for the user
+        memberships = Membership.objects.filter(user=self.request.user)
+        sites_id_list = []
+
+        # get the site ids from the memberships
+        for membership in memberships:
+            sites_id_list.append(membership.site.id)
+
+        # get the site objects filtered by the id list and order by language
+        queryset = (
+            Site.objects.filter(id__in=sites_id_list)
+            .select_related("language")
+            .order_by(Upper("language__title"), Upper("title"))
+        )
+
+        # filter the list down to the sites the user has view permissions on
+        sites = utils.filter_by_viewable(self.request.user, queryset)
+
+        return sites
+
+    def list(self, request, *args, **kwargs):
+        # get the sites
+        sites = self.get_queryset()
+
+        # serialize each site (groupby can't handle Models)
+        site_jsons = [
+            SiteSummarySerializer(site, context={"request": request}).data
+            for site in sites
+        ]
+
+        # group by language
+        rows = groupby(site_jsons, itemgetter("language"))
+        data = [
+            {
+                "language": language if language is not None else "Other",
+                "sites": list(items),
+            }
+            for language, items in rows
+        ]
+
```

Comment:
Can this be extracted to a reusable function that is shared with the SiteViewSet?